src/market/config.ts

```diff
@@ -19,34 +19,51 @@ const DEFAULTS = {
  */
 export class DemandConfig {
   public readonly yagnaOptions?: YagnaOptions;
-  public readonly timeout: number;
   public readonly expiration: number;
   public readonly subnetTag: string;
   public readonly maxOfferEvents: number;
   public readonly offerFetchingInterval: number;
   public readonly logger?: Logger;
   public readonly eventTarget?: EventTarget;
+  public readonly debitNotesAcceptanceTimeoutSec: number;
+  public readonly midAgreementPaymentTimeoutSec: number;
 
   constructor(options?: DemandOptions) {
     this.subnetTag = options?.subnetTag || EnvUtils.getYagnaSubnet() || DEFAULTS.subnetTag;
-    this.timeout = options?.marketTimeout || DEFAULTS.marketTimeout;
-    this.expiration = options?.marketOfferExpiration || DEFAULTS.marketOfferExpiration;
+
+    // Handling demand/activity expiration in a backward compatible way
+    const oldExpiration = options?.marketOfferExpiration;
+    const newExpiration = options?.expiration;
+    this.expiration = oldExpiration || newExpiration || DEFAULTS.expiration;
+
+    if (oldExpiration !== undefined) {
+      this.logger?.warn(
+        "The `DemandOptions.marketOfferExpiration` setting is deprecated, migrate your code to `DemandOptions.expiration`",
+      );
+    }
+
     this.offerFetchingInterval = options?.offerFetchingInterval || DEFAULTS.offerFetchingInterval;
     this.logger = options?.logger;
     this.maxOfferEvents = options?.maxOfferEvents || DEFAULTS.maxOfferEvents;
     this.eventTarget = options?.eventTarget;
+    this.debitNotesAcceptanceTimeoutSec =
+      options?.debitNotesAcceptanceTimeoutSec || DEFAULTS.debitNotesAcceptanceTimeoutSec;
+    this.midAgreementPaymentTimeoutSec =
+      options?.midAgreementPaymentTimeoutSec || DEFAULTS.midAgreementPaymentTimeoutSec;
```

Comment:
Same question as above, setting these settings to `0` will unexpectedly change them to the default values

Reply:
Depending on the setting:

If `debitNotesAcceptanceTimeoutSec` would be set to 0, this would make the demand get rejected by providers or in some cases limit the lifetime of an activity to 30 minutes.

If `midAgreementPaymentTimeoutSec` would be set to 0, this would make the demand get rejected by providers or in some cases limit the lifetime of an activity to 10 hours.

Now, if we will provide the defaults in these cases (like in the other one as well), I think we will be trying to be wrongly "helpful" again.

I think that I'll issue a warning and accept the user setting. I'm only afraid that most of the time the users won't be using logs with the SDK and this might not be noticed.